AzureMonitorAgent/shim.sh

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
+
+# This is the main driver file for AMA extension. This file first checks if Python 2 or 3 is available on the VM 
+# and if yes then uses that Python (if both are available then, default is set to python3) to run extension operations in agent.py
+# Control arguments passed to the shim are redirected to agent.py without validation.
+
+COMMAND="./agent.py"
+PYTHON=""
+FUTURE_PATH=""
+ARG="$@"
+
+function find_python() {
+    local python_exec_command=$1
+    local future_path=$2
+
+    if command -v python2 >/dev/null 2>&1 ; then
+        eval ${python_exec_command}="python2"
+        eval ${future_path}="${PWD}/ext/future:"
+    elif command -v python3 >/dev/null 2>&1 ; then
+        eval ${python_exec_command}="python3"
+        # do not set future_path; future seems to cause interference with preexisting packages in python 3 environment
+    fi
+}
+
+find_python PYTHON FUTURE_PATH
+
+if [ -z "$PYTHON" ] # If python is not installed, we will fail the install with the following error, requiring cx to have python pre-installed
+then
+    echo "No Python interpreter found, which is an AMA extension dependency. Please install either Python 2 or 3." >&2
```

Comment:
My understanding is that eventually we want to be force-installing python3 on the system if we don't find any python. Is that scoped to a separate PR/release?

Reply:
Currently we are inclining towards going with the non-intrusive approach of failing to install if python isn't present on the VM (similar to OMS) rather than force installing python. Need to get PM approval but seems like this current design will likely not change.

Reply:
Gotcha. I was also under the impression that we wanted to prefer python3; in that case `find_python` needs to reflect this since currently it will take python2 if both are available.

And if my impression is correct, maybe the wording on this line could be more targed—for instance, "Please install Python 3, or Python 2 if the former is unavailable."

Reply:
We don't have a preference for which py to go for in the case they are both available, But after the older distros moved to py3, majority of the distros in our support matrix now support py3 so I agree that it makes sense to make that the default